pkg/networkserver/grpc_gsns.go

```diff
@@ -923,8 +924,12 @@ func (ns *NetworkServer) newDevAddr(context.Context, *ttnpb.EndDevice) types.Dev
 
 func (ns *NetworkServer) sendJoinRequest(ctx context.Context, ids ttnpb.EndDeviceIdentifiers, req *ttnpb.JoinRequest) (*ttnpb.JoinResponse, error) {
 	logger := log.FromContext(ctx)
-	if js := ns.GetPeer(ctx, ttnpb.ClusterRole_JOIN_SERVER, ids); js != nil {
-		resp, err := ttnpb.NewNsJsClient(js.Conn()).HandleJoin(ctx, req, ns.WithClusterAuth())
+	var cc *grpc.ClientConn
+	if js, _ := ns.GetPeer(ctx, ttnpb.ClusterRole_JOIN_SERVER, ids); js != nil {
```

Comment:
IDEM

Reply:
OK, so related to the above. Should we:

1. Not bother about an error if the peer is not required
2. Check explicitly for `errors.IsNotFound()` and skip if we don't need it, and return an error on anything else?

Reply:
I vote for 2.